hydro_lang/src/ir.rs

```diff
@@ -1196,6 +1322,40 @@ impl<'a> HydroNode {
             }
         }
     }
+
+    pub fn metadata(&self) -> HydroNodeMetadata {
```

Comment:
This should return an `&HydroNodeMetadata`, which the recipient can chose to clone or not.

Reply:
done